pysoa/server/action/status.py

```diff
@@ -147,6 +169,44 @@ def run(self, request):
 
         return status
 
+    def _check_client_settings(self, request):
+        """
+        This method checks any client settings configured for this service to call other services, calls the `status`
+        action of each configured service with `verbose: False` (which guarantees no further recursive status checking),
+        adds that diagnostic information, and reports any problems. To include this check in your status action, define
+        `check_client_settings = BaseStatusAction._check_client_settings` in your status action class definition.
+        """
+        if not request.client.settings:
+            # There's no need to even add diagnostic details if no client settings are configured
+            return
+
+        self.diagnostics['services'] = {}
+
+        problems = []
+
+        for service_name, _ in six.iteritems(request.client.settings):
+            try:
+                response = request.client.call_action(service_name, 'status', body={'verbose': False})
```

Comment:
There's a subtle bug here, which is that if the remote service is down, this will wait the default configured timeout before raising a MessageSendTimeout or MessageReceiveTimeout exception. The problem is that we tend to set the same default timeout everywhere, so the caller of the healthcheck will timeout too.

Thus, you need to specify a lower timeout here using the new per-request timeout functionality you recently added. I would also suggest that rather than calling each remote service in serial, you use the `call_actions` to call all the remote services in parallel with the same timeout, which I suggest be half of the default timeout things are normally configured with.

Thoughts?

Reply:
All good points.

1) As suggested, I have supplied a lower timeout. It is now 2 seconds. Don't really want to set it any lower.

2) Good idea that this should run in parallel. Unfortunately, the `Client` doesn't yet actually support parallel calling. `call_actions` only supports one service, and it's serial. I'm working on changes now to support parallel calls, but it's not done yet. So, this will have to suffice for now. I will circle back at a later time to improve this.